### DISCUSSION TITLE
Playlist consume mode

### DISCUSSION THREAD

**Discussion by matthiasbeyer** [navidrome/navidrome#3276] [Ideas]
I am used to music playing software that has a "consume mode" for the playlist.
So basically, when I add two albums to the playlist and play all titles, the playlist gets "consumed", meaning that every title that is played is, after it finished, removed from the playlist.

I'd love to see that in navidrome.

**Comment by deluan** [comment]
You can do this easily with a [Smart Playlist](https://github.com/navidrome/navidrome/issues/1417):
```
{
  "all": [
    {"is": {"playCount":0}}
  ],
  "sort": "dateAdded"
}
```

**Reply by matthiasbeyer** [reply]
From what I see, this would only add tracks to the playlist that never have been played, but not remove tracks that have been played from the playlist.

I do _not_ want that I can only have a playlist that only has tracks that never have been played, but a playlist that removes a song as soon as it has been played from that playlist. If I re-add that track after it has been played, I'd assume it to stay there until it has been played again.

**Reply by deluan** [reply]
This is the closest you can get with Smart Playlists, I think:
```
{
  "all": [
    {"lt": {"playCount":1}}
  ],
  "sort": "dateAdded"
}
```

But it does not handle the use case of "re-add that track after it has been played", as you can't add/remove songs directly from a Smart Playlists

**Reply by matthiasbeyer** [reply]
That's why this is a feature request.
